### ISSUE TITLE
Features required that will allow us to leverage camus with AWS auto-scale groups

### ISSUE THREAD

**Issue by jsuresh** [helix-collective/camus#40]
Features should be generic, however, be able to leverage auto-scale groups without us having to 'bake in' anything more than camus into an image.

Even if we don't use auto-scale groups to scale up and down, there's a lot of benefit in having 'fixed' auto-scale groups (as it can also be considered as auto-healing).

An idea (but please discuss more in the issue thread). I'll consolidate into a design doc once the initial flurry of ideas/thoughts are dumped.
   1. camus deploy can be configured to sync to S3 as well. I haven't looked into it too closely, however http://docs.aws.amazon.com/cli/latest/reference/s3/sync.html looks to be pretty similar to rsync.
   2. Set also tweaks some global flag as to which deploy we *expect* to be set as active (using s3 for this seems like the right thing to do). To be general, this could be a JSON file (as we might want to add other data here in the future).
   3. When a machine is started, we configure it with a super-simple cloud-init shell script (in which we configure say, the S3 bucket from which it should pull deploys from). It then pulls 
      a. Pulls say the last 3 deploys
      b. Runs the deploy we *expect* to be active
      c. Sets that deploy as active.

**Comment by jsuresh** [comment]
@soulplant @latos @timbod7 @timcerexhe @barryskal please add comments. I'll move out into a design doc later.

**Comment by latos** [comment]
I don't have any substantial comments yet - let's see the design doc.

Minor thing - I'd rather see as little as possible in cloud-init scripts

**Comment by timbod7** [comment]
> Minor thing - I'd rather see as little as possible in cloud-init scripts

Compared with what? Fully baking the images?

I'd like to see more stuff configured on boot, and as little as possible (ie nothing) configured with ssh later.

**Comment by timbod7** [comment]
@jsuresh - how do you think secrets will be managed?

**Comment by jsuresh** [comment]
Agreed. My plan was to have smartstack style simple bash scripts (just
martial variables, and call python script which gets backed on the machine)

On Sun, 1 May 2016 at 6:34 PM, Daniel Danilatos <notifications@github.com>
wrote:

> I don't have any substantial comments yet - let's see the design doc.
>
> Minor thing - I'd rather see as little as possible in cloud-init scripts
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/helix-collective/camus/issues/40#issuecomment-216025588>
>
-- 

*Jeeva Suresh* | Helix Collective | Founder, Software Engineer

*M:* +61415 671 880 <http://+61415671880/> *A: *Suite 7, Level 2, 13-15
Smail St Ultimo

**Comment by latos** [comment]
@timbod7 as opposed to Camus having options / being configured to pull down a few deploys.  I wasn't suggesting ssh.

**Comment by jsuresh** [comment]
Design doc for camus changes @soulplant @timbod7 @latos @barryskal please comment
https://docs.google.com/document/d/1DgffwF2Lrg1SKqFwz-yKrxvEtbjTCkjk7huVtqmdtqw/edit#
